### DISCUSSION TITLE
TextField htmltext support for subscript and superscript tags (<sub> <sup>)

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#2012] [Ideas]
Is there any chance to make it possible?

**Comment by ajwfrost** [comment]
This looks to be a little challenging to implement as there's no built-in support for these from any of the font rendering methods that we use. So it would be a case of us interpreting this, adding a new font format (smaller) plus a vertical offset (down/up for sub/sup) for these. The new font size part is easy but we'd need to add the vertical offset support into a lot of places, and it's not clear whether we'd then end up with some clipping issues....

We can add this to the backlog but I think it's a lot more effort for us to implement cross-platform than it is for something to just be done at the AS3 level to resize/adjust the text. Potentially we can create a new component within AS3 that's based on a text field that interprets the HTML text to set up a series of text fields where necessary (or, anyone could create this i.e. it just uses current public APIs).

**Reply by MalacTheLittle** [reply]
Series of text fields may be an option for Dynamic `TextField` type, but not sure if it's an option for Input type.
I use `TextField` class for users to edit smaller chunks of text. There are some challenges with bold and italic if I don't use generic device font ("_sans", "_serif", "_typewriter"), but I managed to solve it with `RegExp`. I would use same logic for sub/sup, but the problem is that `TextFormat` doesn't support `baselineShift` property, and it's strange since both Flash Text Engine (FTE) and Text Format Layout (TFL) does.
Only option that is left for us is to create modified version of font in some font edit application and apply it to sub/sup tags with `RegExp` as different font face.

**Reply by ajwfrost** [reply]
Okay yes I see what you mean, and actually having a superscript and subscript version of the font would be quite an 'easy' way to do it although with a lot of overheads... but it would be better to have this handled within the text field itself.

FTE (and a lot of Flex stuff plus the text layout framework) uses a completely different mechanism for rendering, so yes you can do a lot of different things with that. Part of the complexity here is that over the years, there have been so many different ways of doing things - and they're all still present and the actual code path depends on when your content was written, i.e. Adobe kept everything backwards compatible. So any change we make has to be (a) made in a lot of places, and (b) carefully regression-checked!

But we will add it to the to-do list, we were already thinking of updating the font rendering mechanism on Windows so that we can actually support colourful emojis and maybe we'll just do this as a breaking change to simplify it..!

thanks
